src/common/res/features/act-on-change/main.js

```diff
@@ -85,6 +42,72 @@
         if (ynabToolKit.debugNodes) {
           console.log('###')
         }
+        
+        // Loop through the array and act if we find a relevant changed node in the digest
+        for ( var i = 0; i < ynabToolKit.digest.length; i++ ) { 
+
+        	// Changes are detected in the category balances
+        	if ($(ynabToolKit.digest[i]).hasClass("budget-table-cell-available-div")) {
+        	  
+        	  if ( ynabToolKit.featureOptions.updateInspectorColours ) {  
+        	      ynabToolKit.updateInspectorColours();
+        	  }
+
+        	  break;
+        	  
+        	} 
+        }
+
+        for ( var i = 0; i < ynabToolKit.digest.length; i++ ) { 
+
+        	// The user has returned back to the budget screen
+        	if ($(ynabToolKit.digest[i]).hasClass('budget-inspector')) {
+        	  
+        		if ( ynabToolKit.featureOptions.checkCreditBalances ){
```

Comment:
Does it make sense to move the feature flag checking to a single conditional before the loop? These are really lightweight calls, but for a user who has none of these features enabled -- they're doing a lot of extra work.

As a slightly different approach, you could use jQuery's "trigger" and "on" methods for individual features to register themselves for updates. Then the code would check to see if anyone is listening before attempting to filter through the mutations.